### ISSUE TITLE
[Global] Heading structure _may_ be out of order

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#1457]
## Issue summary

In any section instance which includes `h3` headings, heading structure could become out of order if the merchant removes the section "Heading" setting text.

For example, a product detail page with a multicolumn section with no title, the page heading structure includes a primary `h1` then subheading `h3` for each multicolumn item.

This would lead to a confusing user experience when screen reader user navigate via headings alone.

<details>
  <summary>Screenshots</summary>

  <img width="874" alt="Screenshot of a product page with multicolumn section content. Arrows point out the product title as an h1 heading and each multicolumn title as an h3." src="https://user-images.githubusercontent.com/1392632/157051791-2e6ee402-39a8-4134-9b7b-6680be9c2531.png">

</details>

<!-- ### Current code

#### HTML

```html

``` -->

## Steps to reproduce

1. Go to a [Studio demo store PDP page](https://theme-studio-demo.myshopify.com/products/atiyyaat-name-of-urdu-origin-meaning-gifts) with any supported browser
2. With an active screen reader, navigate by headings alone

## Behavior

### Expected

- Section `h2` headings to existing the page.

### Actual

- Section `h2` headings are missing.

## Recommendation

When the merchant removes the section heading text, instead of removing the `h2`:

1. Keep the heading `h2` in the DOM.
2. Apply the `.visually-hidden` class to remove the heading visually
3. Set the `h2` title text as the default text for the section:
   - Featured collection: "Featured collection"
   - Collage: "Multimedia collage"
   - Multicolumn: "Multicolumn"
   - Blog posts: "Blog posts"
   - Collection list: "Collections"

These changes will keep the inteded heading structure intact and inform the user of the section purpose/content type.

```
- <section>
-- section title (h2)
--- item title (h3)
--- item title (h3)
```

### Recommended code

#### HTML

```html
<h2
  class="visually-hidden"
  data-live-text-setting="section.template--15459810640040__1643899285fb6b3031.title/escape"
>
  Multicolumn
</h2>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Dawn version:** 3.0
- **Browser:** Any
- **Component:** Global
- **WCAG Principal:** Operable
- **WCAG SC:** [2.4.10 Section Headings](https://www.w3.org/WAI/WCAG21/Understanding/section-headings.html)
- **Severity:** Medium
- **Effort:** Low
